src/main/java/io/wispforest/owo/serialization/impl/StructField.java

```diff
@@ -0,0 +1,48 @@
+package io.wispforest.owo.serialization.impl;
+
+import io.wispforest.owo.serialization.Endec;
+import io.wispforest.owo.serialization.StructDeserializer;
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public class StructField<S, F> {
+
+    public MutableObject<F> defaultValue = null;
```

Comment:
Why is this a MutableObject?

Reply:
Such acts as a way of telling if a given value is defaulted. Such allows for the default value of null as setting the mutable object to be a instance of what ever defaulted value wanted.

Reply:
What
You mean like it can either be null (required), or non-null, containing the actual default value?

Reply:
Yes, with the default value allowing for either a value of the given type or null